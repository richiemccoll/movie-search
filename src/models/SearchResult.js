class SearchResult {
  constructor(data) {
    this.title = data.title;
    this.release_date = data.release_date;
    this.popularity = data.vote_average;
    this.overview = data.overview;
    this.image_url = this.generateImageUrl(data.poster_path);
  }

  generateImageUrl = (imageSlug) => {
    const baseUrl = "http://image.tmdb.org/t/p/";
    const imageSize = "w342";

    return `${baseUrl}${imageSize}${imageSlug}`;
  };
}

export default SearchResult;
