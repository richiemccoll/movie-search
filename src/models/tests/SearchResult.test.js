import SearchResult from "../SearchResult";

describe("Models - SearchResult", () => {
  it("should generate a model based off data", () => {
    const mockPayload = {
      backdrop_path: "/840rbblaLc4SVxm8gF3DNdJ0YAE.jpg",
      genre_ids: [1, 2, 3],
      id: 332562,
      original_language: "en",
      original_title: "A Star Is Born",
      overview:
        "Seasoned musician Jackson Maine discovers—and falls in love with—struggling artist Ally. She has just about given up on her dream to make it big as a singer—until Jack coaxes her into the spotlight. B…",
      popularity: 192.037,
      poster_path: "/wrFpXMNBRj2PBiN4Z5kix51XaIZ.jpg",
      release_date: "2018-10-03",
      title: "A Star Is Born",
      video: false,
      vote_average: 7.5,
      vote_count: 1079,
    };

    const result = new SearchResult(mockPayload);
    expect(result).toMatchSnapshot();
  });
});
