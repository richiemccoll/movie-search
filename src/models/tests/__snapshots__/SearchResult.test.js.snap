// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Models - SearchResult should generate a model based off data 1`] = `
SearchResult {
  "generateImageUrl": [Function],
  "image_url": "http://image.tmdb.org/t/p/w342/wrFpXMNBRj2PBiN4Z5kix51XaIZ.jpg",
  "overview": "Seasoned musician Jackson Maine discovers—and falls in love with—struggling artist Ally. She has just about given up on her dream to make it big as a singer—until Jack coaxes her into the spotlight. B…",
  "popularity": 7.5,
  "release_date": "2018-10-03",
  "title": "A Star Is Born",
}
`;
